### PR TITLE
Change configuration and data directories

### LINKED ISSUE #231: Add support for XDG directories
e.g. port/rebase and merge, optionally behind a configure flag: https://github.com/audacity/audacity/pull/831
- [X] I have read the specified guidelines for issues

**Is your feature request related to a problem? Please describe.**
The PR noted above seems to be included in multiple Audacity distribution packages. Porting it would make packaging faster and easier.

**Describe the solution you'd like**
e.g. port/rebase and merge, optionally behind a configure flag.

**Describe alternatives you've considered**
Not rebasing+merging it. This would probably winder adoption in multiple distributions.

**Additional context**
Debian and Nixpkgs seem to use the above mentioned PR (via including the associated patch).

### PR DESCRIPTION
**Pull request by nbsp**
Signed-off-by: Campbell Jones <git@serebit.com>
Signed-off-by: Sol Fisher Romanoff <sol@solfisher.com>

Resolves: #231

Port of audacity/audacity#831 by @serebit -- Change the Unix default config directory to `~/.config/audacity`.

Additional things which need to be done:

- [x] Deprecate `.audacity-data` entirely: excusable on Audacity for backwards compatibility, but this is a new program.
- [x] Change the directory names to have Tenacity rather than Audacity.

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Comment by nbsp** [comment]
This works on Windows, where the config&data directory now resides at `AppData\Roaming\tenacity`, however this is what the Build Information tab in the about page shows:

```
Config folder: C:\Users\rancor\AppData\Roaming/tenacity
Data folder:   C:\Users\rancor\AppData\Roaming\Tenacity
```
This is only a visual bug.

File: src/FileNames.cpp [hunk 1]
```diff
@@ -297,26 +278,10 @@ FilePath FileNames::DataDir()
...
-      {
-         // Use old user data dir folder
-         gDataDir = gOldUnixDataDir;
-      } else
-      {
-         wxString dataDir;
-         // see if XDG_DATA_HOME is defined. if it is, use its value. if it isn't, use the default
-         // XDG-specified value
-         if ( !wxGetEnv(wxS("XDG_DATA_HOME"), &dataDir) || dataDir.empty() )
-            dataDir = wxFileName::GetHomeDir() + wxT("/.local/share");
```
Comment (nbsp) [inline src/FileNames.cpp]: Whoops, this sets the datadir to `~/.Tenacity`.

File: src/FileNames.h [hunk 2]
```diff
@@ -112,10 +112,15 @@ namespace FileNames
...
+   /** \brief Audacity user config directory
+    *
+    * Where audacity keeps its settigns squirreled away, by default ~/.config/audacity/
+    * on Unix, Application Data/Audacity on Windows */
+   FilePath ConfigDir();
    /** \brief Audacity user data directory
     *
     * Where audacity keeps its settings and other user data squirreled away,
-    * by default ~/.audacity-data/ on Unix, Application Data/Audacity on
+    * by default ~/.local/share/audacity/ on Unix, Application Data/Audacity on
```
Comment (n0toose) [inline src/FileNames.h]: `~/.local/share/audacity/`?

Comment (nbsp) [reply src/FileNames.h]: Good catch, one sec

Comment (nbsp) [reply src/FileNames.h]: There's a couple of comments I've purposely left unchanged, because I didn't feel they related to this specific issue.

**Comment by nbsp** [comment]
> ```
> Config folder: C:\Users\rancor\AppData\Roaming/tenacity
> Data folder:   C:\Users\rancor\AppData\Roaming\Tenacity
> ```

The problem is now fixed, and the config files are located at `AppData\Roaming\Tenacity`.

**Review by n0toose — APPROVED** [review]
Works on Linux.

**Review by Semisol — APPROVED** [review]
```
Conf: ~/Library/Preferences/tenacity
Data: ~/Library/Application Support/tenacity
```
🎉 Works on MacOS

**Comment by serebit** [comment]
Yay, I helped!

**Comment by lucas-mior** [comment]
Thank you so much!

**Pull request merged**